addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -634,10 +627,14 @@ def message_new(self, msg, custom_values=None):
             'name': msg.get('subject') or _("No Subject"),
             'partner_name': partner_name or email_from_normalized,
         }
-        if msg.get('from') and not self._email_is_blacklisted(msg.get('from')):
+        job_platform = self.env['hr.job.platform'].search([('email', '=', email_from_normalized)], limit=1)
+        if msg.get('from') and not job_platform:
             defaults['email_from'] = msg.get('from')
             defaults['partner_id'] = msg.get('author_id', False)
-        if msg.get('email_from') and self._email_is_blacklisted(msg.get('email_from')):
+        if msg.get('email_from') and job_platform:
+            subject_pattern = re.compile(job_platform.regex)
```

Comment:
Will it work if the regex is not filled?

Reply:
it won't match anything yeah

Reply:
updated it to use an empty string if not filled, it won't match anything